src/components/Message.vue

```diff
@@ -46,10 +46,18 @@
 						<ActionButton icon="icon-mail" @click="onToggleSeen">
 							{{ envelope.flags.unseen ? t('mail', 'Mark read') : t('mail', 'Mark unread') }}
 						</ActionButton>
+						<ActionButton icon="icon-details" @click="onShowSource">
+							{{ t('mail', 'View source') }}
+						</ActionButton>
 						<ActionButton icon="icon-delete" @click="onDelete">
 							{{ t('mail', 'Delete') }}
 						</ActionButton>
 					</Actions>
+					<Modal v-if="showSource" @close="onCloseSource">
```

Comment:
in a follow-up we should extract this into an independent component

Reply:
that's too much for me I believe ;)